### ISSUE TITLE
Feature request: Allow Code Outline in Minimap location

### ISSUE THREAD

**Issue by StevenLangbroek** [microsoft/vscode#50047]
I'm using the VS Code insiders build, which has Code Outline built in. I loved this feature from Nuclide, but in its current incarnation it's not as useful as it could be (to me). I have to choose now between file tree (or git or any other of course) sidebar and Code Outline, but I'd prefer to always have it next to my code.

My idea was to allow what is now the Minimap to use different "widgets" (sorry, I've never developed for VS Code so I don't know the correct terminology), so the Code Outline view could live there instead of the (minimally useful Minimap).

**Comment by StevenLangbroek** [comment]
Oh sorry duplicate. Closing.

**Comment by StevenLangbroek** [comment]
After reading through existing issues I don't think it's a duplicate actually. Feel free to close if you disagree, otherwise I'd like request this feature :).

**Comment by jrieken** [comment]
Yeah, using the minimap location for the outline view is something we have discusses as well. The problem we see with it space, esp. when having multiple editors (think of the newly added grid layout). There might be just a few pixels left and that won't be enough...

**Comment by patrys** [comment]
@jrieken On the other hand, having it where the minimap lives could allow you to compare the structure of multiple files (currently the outline contents switch with focus).

**Comment by henloh** [comment]
I think this would be a great feature. No scrolling to find the function you are looking for. It cound be a setting of some kind or a quick switch button.

**Comment by JerryGoyal** [comment]
that way minimap can actually start providing some value. Much needed feature for better code navigation.

**Comment by sjhsbhqf** [comment]
我也觉得这个功能很棒，不知道是否开发出来了？？

**Comment by andyn** [comment]
> 我也觉得这个功能很棒，不知道是否开发出来了？？

When this issue was first opened, there was no way to have any sidebar panes on the right side of the screen. Nowadays you can open a secondary side bar with Ctrl-Alt-B and drag the outline window there.

For this reason I personally don't need this feature anymore. YMMV.

**Comment by sjhsbhqf** [comment]
Yes, but I still think it would be great if the outline could replace the position of minimap

**Comment by syvuilliot** [comment]
on a large screen, with multiple editors, it should be more confortable to have outline in each editors, instead of needing to put focus on the left editor, then move mouse to the right side of the screen then come back to editor.
And I also dislike that the outline switch at each editor focus. It should be less "flashing" to have outline in each editor.

More generaly, the minimap location could become a "bar" (like the primary and secondary bars) where we could put any menu, no ?
